apps/web/src/hooks/usePaginationLoading.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react'
+
+type Props<T> = {
+  ref: React.RefObject<HTMLDivElement>
+  fetch: () => Promise<T>
+}
+
+export const usePaginationLoading = <T>({ ref, fetch }: Props<T>) => {
+  const [latestHeight, setLatestHeight] = useState<number>(0)
+
+  const handleScroll = async () => {
+    if (
+      ref.current &&
+      latestHeight <= window.scrollY &&
+      latestHeight !== ref.current.offsetHeight
+    ) {
+      setLatestHeight(ref.current.offsetHeight)
+      await fetch()
+    }
+  }
+
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll)
```

Comment:
can we try it with `ref` instead of adding a listener to the window?

something like `ref.current?.addEventListener`

Reply:
@sasicodes I think we have to use `window` because the **scroll** event is not recorded by the `ref` div, only the window.